src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDateFormat = date.split(fromFormat[fromFormat.length - 1]);
+  const newDateFormat = [];
+
+  for (let i = 0; i < oldDateFormat.length; i++) {
+    if (toFormat.indexOf(fromFormat[i]) === -1) {
+      if (fromFormat[i].length === 2) {
+        if (+oldDateFormat[i] < 30) {
+          newDateFormat[toFormat.indexOf('YYYY')] = '20' + oldDateFormat[i];
+        } else {
+          newDateFormat[toFormat.indexOf('YYYY')] = '19' + oldDateFormat[i];
+        }
+      } else {
+        newDateFormat[toFormat.indexOf('YY')] = oldDateFormat[i].slice(-2);
+      }
+      continue;
+    }
```

Comment:
The logic in this loop seems to assume that the `toFormat` array will not contain any new elements that aren't in `fromFormat`, which may not always be the case. The code should be able to handle situations where `toFormat` includes date parts that are not present in `fromFormat`. Furthermore, the loop should not continue if a match is found; it should only skip to the next iteration when a year conversion is performed, which is not clearly handled here.